src/fundus/utils/timeout.py

```diff
@@ -0,0 +1,158 @@
+import _thread as thread
+import contextlib
+import threading
+import time
+from functools import wraps
+from typing import Callable, Iterator, Literal, Optional, TypeVar, overload
+
+from typing_extensions import ParamSpec
+
+P = ParamSpec("P")
+T = TypeVar("T")
+
+
+class Stopwatch:
+    def __init__(self):
+        self._start = time.time()
+
+    @property
+    def time(self) -> float:
+        return max(0.0, time.time() - self._start)
+
+    def reset(self):
+        self._start = time.time()
+
+
+class Timer(threading.Thread):
```

Comment:
I am a bit confused about the naming of the class, since there also is a `threading.Timer` class, which also is referenced in this file and also inherits from `threading.Thread` (refer to other comment). In case the usage of the other implementation was intentional, I would suggest renaming this class to avoid confusion.

Reply:
You're right, the names are confusing. I'm gonna rename my own implementation to `ResettableTImer`.